atomix/cluster/src/main/java/io/atomix/raft/impl/LogCompactor.java

```diff
@@ -69,14 +71,28 @@ public boolean compactIgnoringReplicationThreshold() {
     return compact(compactableIndex);
   }
 
+  /**
+   * Sets the compactable index to the given index; this will cause a call to {@link #compact()} or
+   * {@link #compactIgnoringReplicationThreshold()} to compact the log up to the given index here.
+   *
+   * <p>NOTE: this method is thread safe
+   */
+  public void setCompactableIndex(final long index) {
+    logger.trace("Updated compactable index to {}", index);
+    compactableIndex = index;
+  }
+
+  /** Compacts the log based on the snapshot store's lowest compaction bound. */
+  public void compactFromSnapshots(
```

Comment:
I opted to pass the store/executor as arguments instead of polluting the state of the compactor, but I'm happy to change that if you'd like.